python/ngen_cal/src/ngen/cal/calibration_set.py

```diff
@@ -19,14 +21,16 @@ class CalibrationSet(Evaluatable):
         A HY_Features based catchment with additional calibration information/functionality
     """
 
-    def __init__(self, adjustables: Sequence[Adjustable], eval_nexus: Nexus, routing_output: 'Path', start_time: str, end_time: str, eval_params: 'EvaluationOptions'):
+    def __init__(self, adjustables: Sequence[Adjustable], eval_nexus: Nexus, hooks: 'HookRelay', start_time: str, end_time: str, eval_params: 'EvaluationOptions'):
```

Comment:
Why not just take in a `ModelHooks` instead? Then you get type hints and I think it is just generally more clear what is going on.

Aside, I think the coupling between the `CalibrationX` and `Evaluatable` are starting to show here. Not sure that we we can / should address that here, but I really want to push to try and flatten out the concepts in future additions. I think we will thank ourselves for that in the long run.

Reply:
Looking at the types provided by the plugin manager, I could extract a single hook as a `HookCaller` or all hooks managed by the manager as a `HookRelay` and extract the callers as needed, I opted for the latter to allow more hooks to be added/used as needed.

Reply:
Meta Duck Typing 🫨 will try it out I suppose 😄

Reply:
changed in 2602282227192d6b4703fde0cf155c1a2c003a75